.github/workflows/ci.yaml

```diff
@@ -0,0 +1,155 @@
+        run: |
+          sudo snap connect opensearch:log-observe
+          sudo snap connect opensearch:mount-observe
+          sudo snap connect opensearch:process-control
+          sudo snap connect opensearch:system-observe
+          sudo snap connect opensearch:sys-fs-cgroup-service
+          sudo snap connect opensearch:shmem-perf-analyzer
+
+      - name: Setup and Start OpenSearch
+        run: |
+          # create the certificates
+          sudo snap run opensearch.setup \
+              --node-name cm0 \
+              --node-roles cluster_manager,data \
+              --tls-priv-key-root-pass root1234 \
+              --tls-priv-key-admin-pass admin1234 \
+              --tls-priv-key-node-pass node1234 \
+              --tls-init-setup yes                 # this creates the root and admin certs as well.
+
+          # start opensearch
+          sudo snap start opensearch.daemon
+
+          # wait a bit for it to fully initialize
+          sleep 15s
+
+          # create the security index
+          sudo snap run opensearch.security-init --tls-priv-key-admin-pass=admin1234
+
+      - name: Download snap file for OpenSearch Dashboards
+        uses: actions/download-artifact@v3
+        with:
+          name: opensearch-dashboards_snap_amd64
+          path: .
+
+      - name: Install snap file for Opensearch Dashboards
+        run: |
+          version="$(cat snap/snapcraft.yaml | yq .version)"
+          
+          sudo snap remove --purge opensearch-dashboards
+          sudo snap install opensearch-dashboards_${version}_amd64.snap --dangerous --jailmode
+
+      - name: Test Opensearch Dashboards
+        run: |
+          sudo snap start opensearch-dashboards
+
+          # Wait until the service esteblishes
+          sleep 15
+
+          # Attempt to access the interface using valid authentication
+          authenticate_ok=$(curl -i  -k -XPOST http://localhost:5601/auth/login \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            -H 'osd-xsrf:true' \
+            -d '{"username":"kibanaserver","password":"kibanaserver"}' | grep "200 OK") 
+
+          if [ ! "$authenticate_ok" ]; then
+            exit 1
+          fi
+
+          # Attempt to access the interface wrongly authenticated
+          wrong_credentials_fail=$(curl -i  -k -XPOST http://localhost:5601/auth/login \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            -H 'osd-xsrf:true' \
+            -d '{"username":"admin","password":"test"}' | grep "401 Unauthorized") 
+
+          if [ ! "$wrong_credentials_fail" ]; then
+            exit 1
+          fi
+
+          # Check if logs are collected as expected
+          logfile="/var/snap/opensearch-dashboards/common/var/log/opensearch-dashboards/opensearch_dashboards.log"
+          if ! sudo test -s "${logfile}"; then
+            exit 1
+          fi
```

Comment:
Q: Is sudo here necessary even to check the file size?

Reply:
Sadly yes. I'm not sure under what account the CI tests are run, but it wasn't sufficient to access the logs...

Reply:
More precisely:
![image](https://github.com/canonical/opensearch-dashboards-snap/assets/3271248/67309c90-e59d-4ca6-88b8-862a42b75ca2)

Reply:
Oh, you have read access on the file, but no access into the parent folder I guess. AS you wish!

Reply:
Indeed. 

![image](https://github.com/canonical/opensearch-dashboards-snap/assets/3271248/002f2583-c95f-4997-a449-855746d8ec4a)

TBH I was following Opensearch patterns so I'd just keep them similar, if that's OK...